src/nvim/api/vim.c

```diff
@@ -2451,3 +2451,70 @@ void nvim_del_user_command(String name, Error *err)
 {
   nvim_buf_del_user_command(-1, name, err);
 }
+
+/// Parse command line.
+///
+/// Doesn't check the validity of command or command arguments.
+///
+/// @param str       Command line string to parse. Cannot contain "\n".
+/// @param opts      Optional parameters. Reserved for future use.
+/// @param[out] err  Error details, if any.
+/// @return Dictionary containing command information, with these keys:
+///         - cmd: (string) command name.
+///         - line1: (number) starting line of command range.
+///         - line2: (number) final line of command range.
+///         - force: (boolean) whether command contains a bang.
+///         - args: (array) command arguments.
+Dictionary nvim_parse_cmdline(String str, Dictionary opts, Error *err)
+  FUNC_API_SINCE(10) FUNC_API_FAST
+{
+  Dictionary result = ARRAY_DICT_INIT;
+
+  if (opts.size > 0) {
+    api_set_error(err, kErrorTypeValidation, "opts dict isn't empty");
+    return result;
+  }
+
+  // Parse command line
+  exarg_T ea;
+  char_u *cmdline = vim_strsave((char_u *)str.data);
+
+  if (!parse_cmdline(cmdline, &ea)) {
+    api_set_error(err, kErrorTypeException, "Error while parsing command line");
+    return result;
```

Comment:
`cmdline` is not freed when returning here.

Reply:
Fixed